### PR TITLE
Add `TextInputE`

### LINKED ISSUE #2431: Copy existing `TextInput` to `TextInputE`
> Text fields let users enter new or edit existing single-line text data. 

This change applies to the forms variant and does not include the small variant.

- [x] Copy component
- [x] Apply CSS changes
  - [x] Spacing
  - [x] Icon on left
  - [x] Icon on right
  - [x] Active state
  - [x] Disabled state
  - [x] Error state 

[Design spec](https://docs.google.com/document/d/1V6hTZvG_HZhHixt8ZrC6Yr-_0xzoAR6JoX8zyR2JtUc/edit#heading=h.47nhz8oi1frr)

### PR DESCRIPTION
**Pull request by thyhjwb6**
## Related issue
Closes #2431 

## Overview
Copies the existing `TextInput` to `TextInputE` and applies correct styling.

## Link to preview
https://pr-2459.netlify.app/?path=/story/text-input-experimental--with-label

## Reason
Work for forms v3.

## Work carried out
- [x] Copy
- [x] Apply styling
- [x] Remove not required fields

## Screenshot
Screenshot 2021-06-15 at 11 05 03

## Developer notes
Notes have been added to the board to capture:
- Displaying the error message
- Small size variant
- Clicking the icon sets the focus

### CODE REVIEW

**Comment by k9dh3zij** [comment]
Thanks. Here's some feedback:



- 1. As soon as I click on the Textbox I get the following warning in the console. It doesn't throw any more errors after that, but typing is then constantly laggy for all Textboxes on the page.

`
vendors~main.1bf0c833.iframe.bundle.js:2 Warning: A component is changing an uncontrolled input to be controlled. This is likely caused by the value changing from undefined to a defined value, which should not happen. Decide between using a controlled or uncontrolled input element for the lifetime of the component. More info: https://reactjs.org/link/controlled-components
    at input
    at O (https://pr-2459.netlify.app/vendors~main.1bf0c833.iframe.bundle.js:2:20563)
    at div
    at O (https://pr-2459.netlify.app/vendors~main.1bf0c833.iframe.bundle.js:2:20563)
    at div
    at O (https://pr-2459.netlify.app/vendors~main.1bf0c833.iframe.bundle.js:2:20563)
    at div
    at O (https://pr-2459.netlify.app/vendors~main.1bf0c833.iframe.bundle.js:2:20563)
    at TextInputE (https://pr-2459.netlify.app/main.5d16f1ce.iframe.bundle.js:2:876547)
    at InlineStory (https://pr-2459.netlify.app/vendors~main.1bf0c833.iframe.bundle.js:2:4257139)
    at Story (https://pr-2459.netlify.app/vendors~main.1bf0c833.iframe.bundle.js:2:4258374)
    at MDXProvider (https://pr-2459.netlify.app/vendors~main.1bf0c833.iframe.bundle.js:2:4250323)
    at div
    at Story
    at div
    at div
    at div
    at Styled(div)
    at ZoomElement (https://pr-2459.netlify.app/vendors~main.1bf0c833.iframe.bundle.js:2:1400303)
    at div
    at Styled(div)
    at div
    at Styled(div)
    at div
    at Styled(div)
    at Preview (https://pr-2459.netlify.app/vendors~main.1bf0c833.iframe.bundle.js:2:4278847)
    at MDXProvider (https://pr-2459.netlify.app/vendors~main.1bf0c833.iframe.bundle.js:2:4250323)
    at Canvas (https://pr-2459.netlify.app/vendors~main.1bf0c833.iframe.bundle.js:2:4290171)
    at div
    at Anchor (https://pr-2459.netlify.app/vendors~main.1bf0c833.iframe.bundle.js:2:4183650)
    at DocsStory (https://pr-2459.netlify.app/vendors~main.1bf0c833.iframe.bundle.js:2:4299506)
    at Primary (https://pr-2459.netlify.app/vendors~main.1bf0c833.iframe.bundle.js:2:4300720)
    at DocsPage
    at div
    at Styled(div)
    at div
    at Styled(div)
    at MDXProvider (https://pr-2459.netlify.app/vendors~main.1bf0c833.iframe.bundle.js:2:4250323)
    at ThemeProvider
    at SourceContainer (https://pr-2459.netlify.app/vendors~main.1bf0c833.iframe.bundle.js:2:4283336)
    at DocsContainer (https://pr-2459.netlify.app/vendors~main.1bf0c833.iframe.bundle.js:2:4305198)
printWarning @ vendors~main.1bf0c833.iframe.bundle.js:2
error @ vendors~main.1bf0c833.iframe.bundle.js:2
updateWrapper @ vendors~main.1bf0c833.iframe.bundle.js:2
updateProperties @ vendors~main.1bf0c833.iframe.bundle.js:2
commitUpdate @ vendors~main.1bf0c833.iframe.bundle.js:2
commitWork @ vendors~main.1bf0c833.iframe.bundle.js:2
commitMutationEffects @ vendors~main.1bf0c833.iframe.bundle.js:2
callCallback @ vendors~main.1bf0c833.iframe.bundle.js:2
nrWrapper @ iframe.html?id=text-input-experimental--with-label&viewMode=docs:128
invokeGuardedCallbackDev @ vendors~main.1bf0c833.iframe.bundle.js:2
invokeGuardedCallback @ vendors~main.1bf0c833.iframe.bundle.js:2
commitRootImpl @ vendors~main.1bf0c833.iframe.bundle.js:2
unstable_runWithPriority @ vendors~main.1bf0c833.iframe.bundle.js:2
runWithPriority$1 @ vendors~main.1bf0c833.iframe.bundle.js:2
commitRoot @ vendors~main.1bf0c833.iframe.bundle.js:2
performSyncWorkOnRoot @ vendors~main.1bf0c833.iframe.bundle.js:2
(anonymous) @ vendors~main.1bf0c833.iframe.bundle.js:2
unstable_runWithPriority @ vendors~main.1bf0c833.iframe.bundle.js:2
runWithPriority$1 @ vendors~main.1bf0c833.iframe.bundle.js:2
flushSyncCallbackQueueImpl @ vendors~main.1bf0c833.iframe.bundle.js:2
flushSyncCallbackQueue @ vendors~main.1bf0c833.iframe.bundle.js:2
flushPendingDiscreteUpdates @ vendors~main.1bf0c833.iframe.bundle.js:2
flushDiscreteUpdates @ vendors~main.1bf0c833.iframe.bundle.js:2
finishEventHandler @ vendors~main.1bf0c833.iframe.bundle.js:2
batchedEventUpdates @ vendors~main.1bf0c833.iframe.bundle.js:2
dispatchEventForPluginEventSystem @ vendors~main.1bf0c833.iframe.bundle.js:2
attemptToDispatchEvent @ vendors~main.1bf0c833.iframe.bundle.js:2
dispatchEvent @ vendors~main.1bf0c833.iframe.bundle.js:2
unstable_runWithPriority @ vendors~main.1bf0c833.iframe.bundle.js:2
runWithPriority$1 @ vendors~main.1bf0c833.iframe.bundle.js:2
discreteUpdates$1 @ vendors~main.1bf0c833.iframe.bundle.js:2
discreteUpdates @ vendors~main.1bf0c833.iframe.bundle.js:2
dispatchDiscreteEvent @ vendors~main.1bf0c833.iframe.bundle.js:2
nrWrapper @ iframe.html?id=text-input-experimental--with-label&viewMode=docs:128
Show 4 more frames
`


- 2. Hovering over *and* choosing an option from the browser autocomplete list results in a visual bug

image

image

image



- 3. Chrome reports that the input height is 46px. However when I take a screenshot it seems to actually render at 48px tall. The input height should be 46px tall, including the default border.

image


- 4. Adornment colour should be #3E5667

**Comment by thyhjwb6** [comment]
@k9dh3zij addressed as follows:
- console error was a pre-existing issue which has now been fixed ✅ 
- select from autocomplete visual bug ✅ 
- height of component ✅ 
- adornment color ✅ 
- laggy issue has been raised in a [ticket](https://github.com/Royal-Navy/design-system/issues/2473)

File: packages/react-component-library/src/components/TextInputE/partials/StyledInput.tsx [hunk 1]
```diff
@@ -19,14 +19,22 @@ export const StyledInput = styled.input<StyledInputProps>`
   ${({ $hasLabel }) =>
     $hasLabel &&
     css`
-      padding: ${spacing('10')} ${spacing('6')} ${spacing('2')} ${spacing('6')};
+      padding: ${spacing('10')} ${spacing('6')} ${spacing('2')};
     `}
 
   -webkit-tap-highlight-color: transparent;
+
+  &:-webkit-autofill,
```
Comment (m7kvqbe1) [inline packages/react-component-library/src/components/TextInputE/partials/StyledInput.tsx]: Does this need prefixing for other engines too?

**Review by m7kvqbe1 — CHANGES_REQUESTED** [review]
Code looks good to me but it's bizarre how slow it is when entering text (even in canvas mode in storybook).

Is it worth us creating some tests with react-hook-form? Maybe that's something we do down the line when we have a few more components to play with.

Maybe we should do some manual BrowserStack testing in other browsers?

**Comment by k9dh3zij** [comment]
@thyhjwb6 

Thanks for looking into these. Using Chrome v 91.0.4472.106 I am still getting the visual bug with the input corners when choosing an option from auto-complete (only now they're white rather than light blue). Everything else has been resolved, thank you :) 

image

image

image

**Comment by thyhjwb6** [comment]
@m7kvqbe1

> but it's bizarre how slow it is when entering text

This is a pre-existing issue which has been captured in a [separate ticket](https://github.com/Royal-Navy/design-system/issues/2473)

> Maybe we should do some manual BrowserStack testing in other browsers?

## ie11
ie11

## FF
ff

## Edge
edge


@k9dh3zij thank you and fixed in latest 🙏

**Comment by k9dh3zij** [comment]
All looks good from my side, thank you!

**Review by m7kvqbe1 — CHANGES_REQUESTED** [review]
In your IE11 gif it looks like the label text in the initial state isn't vertically centered.

Maybe it's my eyes?

**Comment by thyhjwb6** [comment]
@m7kvqbe1 it was not your 👀  .. a `fixup` has been applied

## Latest screenshots
### IE 11
Screenshot 2021-06-24 at 11 18 54

### Chrome
Screenshot 2021-06-24 at 11 19 06

**Review by m7kvqbe1 — APPROVED** [review]
👍  LGTM

**Pull request merged**